Bluetooth: controller: Fix channel selection

heml-ot opened the pull request:
For ISO-subevents channel selection must use remap table always -regardless of channel is a mapped or unmapped channel.

Fixes #55736.

Signed-off-by: Henrik Møller <heml@demant.com>

heml-ot committed:
Bluetooth: controller: Fix channel selection

For ISO-subevents channel selection must use remap table always
-regardless of channel is a mapped or unmapped channel.

Signed-off-by: Henrik Møller <heml@demant.com>

cvinayak commented on subsys/bluetooth/controller/ll_sw/lll_chan.c:
```diff
@@ -147,15 +147,9 @@ uint8_t lll_chan_iso_subevent(uint16_t chan_id, uint8_t *chan_map,
 		x = 0;
 	}
 
-	chan_idx = ((((uint32_t)prn_subevent_se * x) >> 16) +
-		    d + *remap_idx) % chan_count;
-
-	if ((chan_map[chan_idx >> 3] & (1 << (chan_idx % 8))) == 0U) {
-		*remap_idx = chan_idx;
-		chan_idx = chan_sel_remap(chan_map, *remap_idx);
-	} else {
-		*remap_idx = chan_idx;
-	}
```
This has been tested by `lll_chan_sel_2_ut()` with the sample data in the specification. Please add a testcase that fails with the implementation.

heml-ot replied:
Yes, I know it’s tested against reference data from spec. Unfortunately, the sample data in spec. is poorly chosen. For a mapped ch; as long as the ch_idx and the remap_idx are identical the issue will not reveal itself. We need a different ch map to trigger the issue.
Point taken; I’ll provide a test catching the issue.

cvinayak replied:
I have added 2 used channels channel map to cover the change.

cvinayak committed:
Bluetooth: Controller: Add additional LE Channel Selection tests

Add additional custom LE Channel Selection #2 tests to cover
event and subevent mapping.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak committed:
Bluetooth: Controller: Add 'U' suffix when using unsigned variables

Add a 'U' suffix to values, when computing and comparing
against unsigned variables.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak committed:
Bluetooth: Controller: Remove redundant use of local variable

Remove redundant use of local variable chan_idx in the
lll_chan_iso_subevent() function.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak commented:
@heml-ot and @wopu-ot can someone from your end approve this?

heml-ot reviewed (commented):
yes, this will work for me.

wopu-ot reviewed (approved):
LGTM
